src/Utility/Text.php

```
@@ -1037,7 +1037,7 @@ public static function parseFileSize(string $size, mixed $default = false): mixe
             return (int)$size;
         }
 
-        if ($default !== false) {
+        if ($default !== null) {
             return $default;
```

We could cast the default to int and keep the return value of the method as `int` only.

---

I just thought someone is using it to find out if the parsing worked. And 0 wouldnt be clear i guess

---

In that case we can make the return type nullable int?

---

hmm.. the docblock has e..g for using string for invalid size, `Value to be returned when invalid size was used, for example 'Unknown type'`

---

I would rather just throw an exception for un-parseable inputs

---

You can add your changes on top if you want to.

---

I updated the return type annotation which avoids having to cast the return value to `int` each time. So I think no additional change is necessary.